fix: remove features depending on DHT sensor type

JKRhb opened the pull request:
https://github.com/RIOT-OS/RIOT/pull/16934 simplified and fixed the driver for DHT sensors. However, by doing so, it removed the possibility to distinguish between the different sensor types (e. g. DHT11 and DHT22). This PR therefore removes all code that depends on this distinction, unfortunately reducing the feature set a litte.

If the PR managed to fix the issue, then the CI pipeline should simply pass in contrast to before (see the latest commit on the main branch).

JKRhb committed:
fix: remove features depending on DHT sensor type

JKRhb committed:
Update RIOT

JKRhb commented:
@schulztr Hmm, the current state of `wot_gcoap` does not support Thing Models anymore :/ Not sure how the fix it at the moment

schulztr commented:
> @schulztr Hmm, the current state of `wot_gcoap` does not support Thing Models anymore :/ Not sure how the fix it at the moment

Oh I'm sorry I think I broke something at the command parsing. As I can see you already fixed that. Furthermore there was a bug when using plain resources or secure resources only. I opened a [PR](https://github.com/namib-project/RIOT/pull/31) which should fix that.

schulztr commented:
The current version of `wot_coap` checks whether `@type` is equal to `tm:ThingModel` to determine whether the given file is a Thing Model or a Thing Description. So I think the Thing Models describing the sensors also need the `@Type`-prefix `tm:`.

schulztr commented on sensor_nodes/wot-config/bindings.json:
```diff
@@ -81,33 +81,6 @@
                 }
             ]
         },
-        "temperature_range": {
```
I'm not sure, but I think this results in an error because the form for  `temperature_range` is missing.

JKRhb replied:
Hmm, I think this should not be a problem as affordances that do not have a form in the consolidated version *should* be ignored.

JKRhb commented:
@schulztr Awesome, thank you :)

JKRhb commented:
@schulztr Hmm, unfortunately, the CI is still failing.

schulztr commented:
> @schulztr Hmm, unfortunately, the CI is still failing.

@JKRhb Sorry for the late response. If the Thing Models don't have the `@Type`-prefix `tm:` the Thing Model is handled as a Thing Description and an Error occurs. Otherwise the `generate_coap.py`-Script failes with the error message

```
line 181, in generate_coap_resources
    forms: List[dict] = affordance["forms"]
KeyError: 'forms'
```
If you print out the affordance name it appears that `temperature_range` is the affordance, which causes this error. This is why i thought that the missing form might be a problem but maybe we could just don't add an affordance to the resource list in line 182 if it don't have a form.

JKRhb committed:
fix @type of Thing Models

JKRhb committed:
fix tm:required definitions

JKRhb commented:
> > @schulztr Hmm, unfortunately, the CI is still failing.
> 
> @JKRhb Sorry for the late response. If the Thing Models don't have the `@Type`-prefix `tm:` the Thing Model is handled as a Thing Description and an Error occurs. Otherwise the `generate_coap.py`-Script failes with the error message
> 
> ```
> line 181, in generate_coap_resources
>     forms: List[dict] = affordance["forms"]
> KeyError: 'forms'
> ```
> 
> If you print out the affordance name it appears that `temperature_range` is the affordance, which causes this error. This is why i thought that the missing form might be a problem but maybe we could just don't add an affordance to the resource list in line 182 if it don't have a form.

Thank you for pointing out these issues! I prefixed the Thing Model `@type` now :) Regarding the `temperature_range` issue: There is a mechanism in place in the generation script for determining required properties and leaving them out if they do not fulfill the requirements, but this is not working correctly. I will therefore remove the affordance for now and create an issue as a reminder that this needs to be addressed.

JKRhb committed:
fix: remove unexpanded affordances

As there is currently a bug with the tm:required feature,
these affordances need to be removed in order to let the
application still compile.
